src/services.py

```diff
@@ -143,3 +157,52 @@ def __init__(self, Security=Security):
     def get_all(self):
         with session_scope() as session:
             return [sec.asdict() for sec in session.query(self.Security).all()]
+
+
+class LinkedinService:
+    def __init__(self, User=User):
+        self.User = User
+
+    @validate_input(LINKEDIN_CODE_SCHEMA)
+    def get_user_data(self, code, redirect_uri):
+        token = self.get_token(code=code, redirect_uri=redirect_uri)
+        return self.get_user_email(**token)
+
+    @validate_input(LINKEDIN_CODE_SCHEMA)
+    def get_token(self, code, redirect_uri):
+        token = requests.post(
+            "https://www.linkedin.com/oauth/v2/accessToken",
+            headers={"Content-Type": "x-www-form-urlencoded"},
+            params={
+                "grant_type": "authorization_code",
+                "code": code,
+                "redirect_uri": redirect_uri,
+                "client_id": APP_CONFIG.get("CLIENT_ID"),
+                "client_secret": APP_CONFIG.get("CLIENT_SECRET"),
+            },
+        ).json()
+        return {"token": token.get("access_token")}
+
+    @validate_input(LINKEDIN_TOKEN_SCHEMA)
+    def get_user_profile(self, token):
+        user_profile = requests.get(
+            "https://api.linkedin.com/v2/me",
+            headers={"Authorization": f"Bearer {token}"},
+        ).json()
+        first_name = user_profile.get("localizedFirstName")
+        last_name = user_profile.get("localizedLastName")
+        return {"full_name": f"{first_name} {last_name}"}
+
+    @validate_input(LINKEDIN_TOKEN_SCHEMA)
+    def get_user_email(self, token):
+        email = requests.get(
+            "https://api.linkedin.com/v2/emailAddress?q=members&projection=(elements*(handle~))",
+            headers={"Authorization": f"Bearer {token}"},
+        ).json()
+        return {"email": email.get("elements")[0].get("handle~").get("emailAddress")}
+
+    @validate_input(LINKEDIN_MATCH_EMAILS_SCHEMA)
+    def is_email_matching(self, user_email, linkedin_email):
```

Comment:
If the method name is called `is_*` I would expect this to return a boolean value. Can we just make this return `user_email = linkedin_email` and then raise the exception accordingly in the function that uses this?

In fact I think it is better if you just inline this lah.